FilteringApp/FilteringApp.Core/Utils/ArrayUtils.cs

```diff
@@ -0,0 +1,96 @@
+﻿using FilteringApp.Core.Models;
+using System;
+using System.Collections.Generic;
+
+
+namespace FilteringApp.Core.Utils
+{
+    public static class ArrayUtils
+    {
+        // do not change input parameter values, like NEVER!!!
+        public static List<int> FilterOutResults(UserInput input)
+        {
+            var result = new List<int>();
+
+            foreach (int v in input.UserArray)
+            {
+                // and not changing is here below! IMPORTANT!
+                var current = v;
```

Comment:
If you name 'v' better, you can get rid off this extra variable and use result.Add(myBetterName) :)

Reply:
Actually not, because at the line 23 we do "current /= 10" and that wouldn't be possible, since you can't change value of a "foreach" variable ;) And even if we could, that would cause change of values in the array, which will not be the best option, because caller may not expect that. That't why we need a guard value.